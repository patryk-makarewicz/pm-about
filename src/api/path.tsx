// https://api.github.com/users/patryk-makarewicz/repos?sort=created
