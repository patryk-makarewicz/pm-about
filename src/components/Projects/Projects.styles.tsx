import styled from 'styled-components';
import { Carousel as AntdCarousel } from 'antd';
import { colors } from 'styles';

export const ItemContainer = styled.div`
  height: 200px;
  color: ${colors.white};
  line-height: 160px;
  text-align: center;
  background: ${colors.black};
`;

export const Carousel: typeof AntdCarousel = styled(AntdCarousel)`
  .slick-dots > li > button {
    background-color: ${colors.green};
  }
  .slick-dots > li.slick-active > button {
    background-color: ${colors.green};
  }

  .ant-carousel,
  .slick-slider,
  .slick-list {
    border-radius: 8px;
  }
`;

export const CarouselContainer = styled.div`
  max-width: 500px;
  margin: 40px auto 0;
`;

export const Paragraph = styled.p`
  color: var(--text-color);
  text-align: center;
`;
