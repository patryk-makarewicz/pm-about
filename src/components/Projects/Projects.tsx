import { Carousel } from 'antd';
import { useTranslation } from 'react-i18next';
import * as Styled from './Projects.styles';
import React from 'react';
import { useRepoList } from 'hooks/useRepoList';

const contentStyle: React.CSSProperties = {
  height: '160px',
  color: '#fff',
  lineHeight: '160px',
  textAlign: 'center',
  background: '#364d79'
};

export const Projects = () => {
  const { t } = useTranslation();

  const { data: repoList } = useRepoList();

  console.log(repoList);

  return (
    <div id="projects">
      <Styled.SectionTitle>{t('projects.title')}</Styled.SectionTitle>
      <Styled.Paragraph>{t('projects.description')}</Styled.Paragraph>
      <Carousel autoplay>
        <div>
          <h3 style={contentStyle}>1</h3>
        </div>
        <div>
          <h3 style={contentStyle}>2</h3>
        </div>
        <div>
          <h3 style={contentStyle}>3</h3>
        </div>
        <div>
          <h3 style={contentStyle}>4</h3>
        </div>
      </Carousel>
    </div>
  );
};
