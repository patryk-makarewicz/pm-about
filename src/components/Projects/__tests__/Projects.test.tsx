import { render, cleanup, waitFor } from '@testing-library/react';
import { queryClient } from 'api';
import { QueryClientProvider } from 'react-query';

import { Projects } from '../Projects';

afterEach(cleanup);

const renderProjects = () => {
  const onClick = jest.fn();
  const { baseElement, container, getByText } = render(
    <QueryClientProvider client={queryClient}>
      <Projects />
    </QueryClientProvider>
  );

  return {
    baseElement,
    onClick,
    container,
    getByText
  };
};

describe('Testing Projects Component', () => {
  test('should take snapshot', async () => {
    const { baseElement } = renderProjects();

    await waitFor(() => expect(baseElement).toMatchSnapshot());
  });
});
