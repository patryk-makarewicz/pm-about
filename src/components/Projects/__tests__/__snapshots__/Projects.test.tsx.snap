// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Projects Component should take snapshot 1`] = `
.c0 {
  color: var(--text-color);
  font-size: 1.3125rem;
  margin-bottom: 30px;
  text-align: center;
  font-weight: 600;
}

.c2 {
  max-width: 300px;
  margin: 40px auto 0;
}

.c1 {
  color: var(--text-color);
  text-align: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

<body>
  <div>
    <div
      id="projects"
    >
      <h2
        class="c0"
      >
        projects.title
      </h2>
      <p
        class="c1"
      >
        projects.description
      </p>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            aria-busy="true"
            aria-live="polite"
            class="ant-spin ant-spin-spinning"
          >
            <span
              aria-label="loading"
              class="anticon anticon-loading anticon-spin ant-spin-dot"
              role="img"
              style="font-size: 36px;"
            >
              <svg
                aria-hidden="true"
                data-icon="loading"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
